homeassistant/components/switch/unifi.py

```diff
@@ -0,0 +1,218 @@
+                         switches[client_id].client.mac)
+            switches[client_id].async_schedule_update_ha_state()
+            continue
+
+        client = controller.api.clients[client_id]
+        if not client.is_wired or client.sw_mac not in devices or \
+           not devices[client.sw_mac].ports[client.sw_port].port_poe or \
+           not devices[client.sw_mac].ports[client.sw_port].poe_enable:
+            continue
+
+        switches[client_id] = UniFiSwitch(
+            client, controller, request_controller_update)
+        new_switches.append(switches[client_id])
+        LOGGER.debug("New UniFi switch %s (%s)", client.hostname, client.mac)
+
+    if new_switches:
+        async_add_entities(new_switches)
+
+
+class UniFiSwitch(SwitchDevice):
+    """Representation of a client that uses POE."""
+
+    def __init__(self, client, controller, request_controller_update):
+        """Set up switch."""
+        self.client = client
+        self.controller = controller
+        self.poe_mode = None
+        if self.port.poe_mode != 'off':
+            self.poe_mode = self.port.poe_mode
+        self.async_request_controller_update = request_controller_update
+
+    async def async_update(self):
+        """Synchronize state with controller."""
+        await self.async_request_controller_update(self.client.mac)
+
+    @property
+    def name(self):
+        """Return the name of the switch."""
+        return self.client.hostname
+
+    @property
+    def unique_id(self):
+        """Return a unique identifier for this switch."""
+        return 'poe-{}'.format(self.client.mac)
+
+    @property
+    def is_on(self):
+        """Return true if POE is active."""
+        return self.port.poe_mode != 'off'
+
+    @property
+    def available(self):
+        """Return if switch is available."""
+        return self.controller.available or \
+            self.client.sw_mac in self.controller.api.devices
+
+    async def async_turn_on(self, **kwargs):
+        """Enable POE for client."""
+        await self.device.async_set_port_poe_mode(
+            self.client.sw_port, self.poe_mode)
+
+    async def async_turn_off(self, **kwargs):
+        """Disable POE for client."""
+        await self.device.async_set_port_poe_mode(self.client.sw_port, 'off')
+
+    @property
+    def device_state_attributes(self):
+        """Return the device state attributes."""
+        attributes = {
+            'power': self.port.poe_power,
+            'received': self.client.wired_rx_bytes / 1000000,
+            'sent': self.client.wired_tx_bytes / 1000000,
+            'switch': self.client.sw_mac,
+            'port': self.client.sw_port,
+            'poe_mode': self.poe_mode
+        }
+        return attributes
+
+    @property
+    def device_info(self):
+        """Return a device description for device registry."""
+        return {
+            'connections': {(CONNECTION_NETWORK_MAC, self.client.mac)},
+            'name': self.name,
```

Comment:
This might be tricky. If another integration will also report the name, we are going to override it with the hostname. The actual integration probably has a better name for the device.